### PR TITLE
Updated User Profile Service by adding Create and Get User Preference

### PR DESCRIPTION
**Pull request by leobcgov**
## Status
**Ready**

## Fixes or Implements [AB#8627](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/8627)
- [X] Enhancement
- [ ] Bug
- [ ] Documentation

## Description:
- Update User Profile Service by adding Create and Get User Preference.
- Include Unit Test for the service.

## Testing
- [X] Unit Tests Updated
- [ ] Functional Tests Updated
- [ ] Not Required

### Steps to Reproduce:
N/A

### UI Changes
NO

### Browsers Tested
- [ ] Chrome
- [ ] Safari
- [ ] Edge
- [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments
Any addtional notes or comments that may be relevant.  Include any specialized deployment steps here.

### CODE REVIEW

File: Apps/Database/src/Models/UserPreference.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+        /// <summary>
+        /// Gets or sets the user hdid.
+        /// </summary>
+        [Key]
+        [Column("UserProfileId")]
+        [MaxLength(52)]
+        public string HdId { get; set; } = null!;
+
+        public bool DismissedMyNotePopover { get; set; }
```
Comment (tiagograf-AOT) [inline Apps/Database/src/Models/UserPreference.cs:37]: missing documentation?!

File: Apps/Database/src/Delegates/IPreferenceDelegate.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+{
+    using System;
+    using System.Collections.Generic;
+    using HealthGateway.Database.Models;
+    using HealthGateway.Database.Wrapper;
+
+    /// <summary>
+    /// Delegate that performs operations for the UserPreference model.
+    /// </summary>
+    public interface IPreferenceDelegate
```
Comment (FuriousLlama) [inline Apps/Database/src/Delegates/IPreferenceDelegate.cs:26]: Kinda nit: IUserPreferenceDelegate

File: Apps/Database/src/Models/UserPreference.cs (hunk 1 above)
Comment (FuriousLlama) [inline Apps/Database/src/Models/UserPreference.cs:37]: Not 100% this would be a good idea since for each parameter we will have to create a column

Comment (leobcgov) [reply Apps/Database/src/Models/UserPreference.cs:37]: I don't see any documentations in the other db models. Can you guys confirm?

File: Apps/WebClient/src/Server/Models/UserPreferenceModel.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+// limitations under the License.
+//-------------------------------------------------------------------------
+namespace HealthGateway.WebClient.Models
+{
+    using System;
+
+    /// <summary>
+    /// Model that provides a user representation of an user preference database model.
+    /// </summary>
+    public class UserPreferenceModel
```
Comment (FuriousLlama) [inline Apps/WebClient/src/Server/Models/UserPreferenceModel.cs:23]: Clases like these are only necessary if the db model is different from the one requested at the controller level

File: Apps/Database/src/Models/UserPreference.cs (hunk 1 above)
Comment (FuriousLlama) [inline Apps/Database/src/Models/UserPreference.cs:37]: This one we need to chat about

**Pull request merged**